### ISSUE TITLE
Implement handling of Expected exceptions in Doctests.

### ISSUE THREAD

**Issue by Carreau** [jupyter/papyri#274]
Currently we just exclude docstrings examples which raise exceptions unconditionally. 

We should check that the error match what is in the docstring, like the classic doctes.

**Comment by asmeurer** [comment]
If papyri is going to execute doctest examples it should just use the `doctest` module directly.

**Comment by Carreau** [comment]
The problem is that we execute also for Jedi to properly infer which objects are in example. 
It's not present right now, but we used to have cross links for every token in code example. 
We also use that to capture the created images by matplotlib, and display them

I was unable to do it with doctest.

**Comment by asmeurer** [comment]
In my experience doctest is pretty extensible. I think it might be possible to hook into it to handle these things.

**Comment by melissawm** [comment]
May be of interest: https://github.com/ev-br/scpdt

**Comment by asmeurer** [comment]
Really we need to make the whole doctester pluggable, so that different libraries can use different doctesting tools. Picking just one thing as a default is in appropriate for all libraries, although if there has to be a default, it should be the basic `doctest`.

**Comment by Carreau** [comment]
I agree that we should make it pluggable, like many other things, but this might come at a later date, once we have a better understanding of what the needs are.
